src/VisualStudio/Core/Def/ColorSchemes/ColorSchemeApplier.cs

```diff
@@ -142,33 +129,36 @@ private bool TryGetUpdatedColorScheme([NotNullWhen(returnValue: true)] out Color
                 : ColorSchemeName.VisualStudio2017;
 
             if (appliedColorScheme == configuredColorScheme)
-            {
-                colorScheme = null;
-                return false;
-            }
+                return null;
 
-            colorScheme = configuredColorScheme;
-            return true;
+            return configuredColorScheme;
         }
 
         public bool IsSupportedTheme()
-            => IsSupportedTheme(GetThemeId());
+            => _threadingContext.JoinableTaskFactory.Run(() => IsSupportedThemeAsync(_threadingContext.DisposalToken));
```

Comment:
Use of _threadingContext.DisposalToken seems a bit odd here, since it means whoever is calling IsSupportedTheme() might get it potentially cancelled out from underneath them. Can the caller pass through a token?